PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = ["account.journal"]
-    _inherit = ['account.move']
-    _inherit = ['account.payment']
-    _inherit = ['account.payment.method']
-    _inherit = ["account.payment.method.line"]
-    _inherit = ['onboarding.onboarding.step']
-    _inherit = ['payment.provider']
-    _inherit = ['payment.transaction']

Total Changes: 16

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/account_payment/models/account_journal.py
+++ b/addons/account_payment/models/account_journal.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class AccountJournal(models.Model):
-    _inherit = ["account.journal"]
+    _inherit = "account.journal"
 
     def _get_available_payment_method_lines(self, payment_type):
         lines = super()._get_available_payment_method_lines(payment_type)

--- a/addons/account_payment/models/account_move.py
+++ b/addons/account_payment/models/account_move.py
@@ -7,7 +7,7 @@ from odoo.addons.payment import utils as payment_utils
 
 
 class AccountMove(models.Model):
-    _inherit = ['account.move']
+    _inherit = 'account.move'
 
     transaction_ids = fields.Many2many(
         string="Transactions", comodel_name='payment.transaction',

--- a/addons/account_payment/models/account_payment.py
+++ b/addons/account_payment/models/account_payment.py
@@ -5,7 +5,7 @@ from odoo.exceptions import ValidationError
 
 
 class AccountPayment(models.Model):
-    _inherit = ['account.payment']
+    _inherit = 'account.payment'
 
     # == Business fields ==
     payment_transaction_id = fields.Many2one(

--- a/addons/account_payment/models/account_payment_method.py
+++ b/addons/account_payment/models/account_payment_method.py
@@ -5,7 +5,7 @@ from odoo import api, models
 
 
 class AccountPaymentMethod(models.Model):
-    _inherit = ['account.payment.method']
+    _inherit = 'account.payment.method'
 
     @api.model
     def _get_payment_method_information(self):

--- a/addons/account_payment/models/account_payment_method_line.py
+++ b/addons/account_payment/models/account_payment_method_line.py
@@ -7,7 +7,7 @@ from odoo.osv import expression
 
 
 class AccountPaymentMethodLine(models.Model):
-    _inherit = ["account.payment.method.line"]
+    _inherit = "account.payment.method.line"
 
     payment_provider_id = fields.Many2one(
         comodel_name='payment.provider',

--- a/addons/account_payment/models/onboarding_onboarding_step.py
+++ b/addons/account_payment/models/onboarding_onboarding_step.py
@@ -4,7 +4,7 @@ from odoo import api, models
 
 
 class OnboardingOnboardingStep(models.Model):
-    _inherit = ['onboarding.onboarding.step']
+    _inherit = 'onboarding.onboarding.step'
 
     @api.model
     def action_open_step_payment_provider(self):

--- a/addons/account_payment/models/payment_provider.py
+++ b/addons/account_payment/models/payment_provider.py
@@ -5,7 +5,7 @@ from odoo.exceptions import UserError
 
 
 class PaymentProvider(models.Model):
-    _inherit = ['payment.provider']
+    _inherit = 'payment.provider'
 
     journal_id = fields.Many2one(
         string="Payment Journal",

--- a/addons/account_payment/models/payment_transaction.py
+++ b/addons/account_payment/models/payment_transaction.py
@@ -4,7 +4,7 @@ from odoo import SUPERUSER_ID, _, api, fields, models
 
 
 class PaymentTransaction(models.Model):
-    _inherit = ['payment.transaction']
+    _inherit = 'payment.transaction'
 
     payment_id = fields.Many2one(
         string="Payment", comodel_name='account.payment', readonly=True)
